### PR TITLE
feat: scope order & order items, cleanup

### PR DESCRIPTION
**Pull request by epuluikis**

### CODE REVIEW

**Review by BaltasRokis — APPROVED** [review]

**Review by titasstank — CHANGES_REQUESTED** [review]

File: Looms.PoS/Looms.PoS.Application/Features/OrderItem/Commands/UpdateOrderItem/UpdateOrderItemRequestValidator.cs [hunk 1]
```diff
@@ -10,38 +11,98 @@ public class UpdateOrderItemRequestValidator : AbstractValidator<UpdateOrderItem
...
+            .CustomAsync(async (productId, context, _) =>
             {
-                var discount = await discountsRepository.GetAsync(Guid.Parse(discountId));
+                var productDao = await productsRepository.GetAsyncByIdAndBusinessId(
+                    Guid.Parse(productId!),
+                    Guid.Parse((string)context.RootContextData[HeaderConstants.BusinessIdHeader])
+                );
 
-                if (discount.Target == DiscountTarget.Order)
+                if (productDao.Quantity < context.InstanceToValidate.Quantity)
```
Comment (titasstank) [inline Looms.PoS/Looms.PoS.Application/Features/OrderItem/Commands/UpdateOrderItem/UpdateOrderItemRequestValidator.cs:43]: should check if the increase in quantity is more than the available stock

File: Looms.PoS/Looms.PoS.Application/Features/OrderItem/Commands/UpdateOrderItem/UpdateOrderItemRequestValidator.cs [hunk 2]
```diff
@@ -10,38 +11,98 @@ public class UpdateOrderItemRequestValidator : AbstractValidator<UpdateOrderItem
...
             {
-                var variation = await variationsRepository.GetAsync(Guid.Parse(variationId));
+                var variation =
+                    await variationRepository.GetAsyncByIdAndProductId(
+                        Guid.Parse(variationId!),
+                        Guid.Parse(context.InstanceToValidate.ProductId!)
+                    );
 
-                if (variation.Quantity < context.InstanceToValidate.Quantity)
+                if (variation!.Quantity < context.InstanceToValidate.Quantity)
```
Comment (titasstank) [inline Looms.PoS/Looms.PoS.Application/Features/OrderItem/Commands/UpdateOrderItem/UpdateOrderItemRequestValidator.cs]: should check if the increase in quantity is more than the available stock

File: Looms.PoS/Looms.PoS.Application/Features/OrderItem/Commands/UpdateOrderItem/UpdateOrderItemRequestValidator.cs [hunk 3]
```diff
@@ -10,38 +11,98 @@ public class UpdateOrderItemRequestValidator : AbstractValidator<UpdateOrderItem
...
+
+                if (discount.Target == DiscountTarget.Order)
+                {
+                    context.AddFailure("Discount cannot be applied to order item.");
+
+                    return;
+                }
+
+                if (context.InstanceToValidate.ProductId is not null
+                 && discount.ProductId != Guid.Parse(context.InstanceToValidate.ProductId))
```
Comment (titasstank) [inline Looms.PoS/Looms.PoS.Application/Features/OrderItem/Commands/UpdateOrderItem/UpdateOrderItemRequestValidator.cs]: will throw if product/service is invalid guid and discount is valid non order type

File: Looms.PoS/Looms.PoS.Application/Looms.PoS.Application.csproj [hunk 4]
```diff
@@ -7,25 +7,26 @@
...
     <ItemGroup>
-        <ProjectReference Include="..\Looms.PoS.Domain\Looms.PoS.Domain.csproj" />
+        <ProjectReference Include="..\Looms.PoS.Domain\Looms.PoS.Domain.csproj"/>
     </ItemGroup>
 
     <ItemGroup>
-        <Folder Include="Features\Payment\Handlers\" />
+        <Folder Include="Features\Payment\Handlers\"/>
+        <Folder Include="Services\Product\"/>
     </ItemGroup>
```
Comment (titasstank) [inline Looms.PoS/Looms.PoS.Application/Looms.PoS.Application.csproj:30]: left over empty folders

File: Looms.PoS/Looms.PoS.Application/Mappings/ModelsResolvers/OrderItemModelsResolver.cs [hunk 5]
```diff
@@ -12,48 +12,80 @@ namespace Looms.PoS.Application.Mappings.ModelsResolvers;
...
+        );
 
-        var taxDao = productDao is not null ? productDao.Tax : serviceDao?.Tax;
-        var tax = _orderItemTotalsService.CalculateOrderItemTax(taxDao, price);
+        var tax = 0m;
 
-        return _mapper.Map<OrderItemDao>(createOrderItemRequest) with
+        var taxDao = productDao?.Tax ?? serviceDao?.Tax;
+
+        if (taxDao != null)
```
Comment (titasstank) [inline Looms.PoS/Looms.PoS.Application/Mappings/ModelsResolvers/OrderItemModelsResolver.cs:81]:
```suggestion
        if (taxDao is not null)
```

File: Looms.PoS/Looms.PoS.Application/Services/OrderService.cs [hunk 6]
```diff
@@ -4,101 +4,137 @@
...
-        _orderItemTotalsService = orderItemTotalsService;
-        _discountTotalsService = discountTotalsService;
+        _orderItemService = orderItemService;
+        _discountService = discountService;
     }
 
-    public decimal CalculateOrderTotal(OrderDao order)
+    public decimal CalculateTotal(OrderDao order)
     {
         decimal total = 0;
```
Comment (titasstank) [inline Looms.PoS/Looms.PoS.Application/Services/OrderService.cs:20]:
```suggestion
        var total = 0m;
```

File: Looms.PoS/Looms.PoS.Application/Services/OrderItemService.cs [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+        return price;
+    }
+
+    public decimal CalculateOrderItemPrice(
+        ProductDao? productDao,
+        ProductVariationDao? productVariationDao,
+        ServiceDao? serviceDao,
+        int quantity)
+    {
+        decimal price = 0;
```
Comment (titasstank) [inline Looms.PoS/Looms.PoS.Application/Services/OrderItemService.cs:46]:
```suggestion
        var price = 0m;
```

File: Looms.PoS/Looms.PoS.Application/Mappings/ModelsResolvers/OrderItemModelsResolver.cs [hunk 8]
```diff
@@ -12,48 +12,80 @@ namespace Looms.PoS.Application.Mappings.ModelsResolvers;
...
-        var taxDao = orderItemDao.Product is not null ? orderItemDao.Product.Tax : orderItemDao.Service?.Tax;
-        var tax = _orderItemTotalsService.CalculateOrderItemTax(taxDao, price);
+        var price = _orderItemService.CalculateOrderItemPrice(
+            productDao,
+            productVariationDao,
+            serviceDao,
+            createOrderItemRequest.Quantity
+        );
+
+        decimal tax = 0;
```
Comment (titasstank) [inline Looms.PoS/Looms.PoS.Application/Mappings/ModelsResolvers/OrderItemModelsResolver.cs:40]:
```suggestion
        var tax = 0m;
```

File: Looms.PoS/Looms.PoS.Application/Services/RefundService.cs [hunk 9]
```diff
@@ -3,12 +3,13 @@
 
 namespace Looms.PoS.Application.Services;
 
-public class RefundsTotalsService : IRefundsTotalsService
+public class RefundService : IRefundService
 {
     public decimal CalculateRefundTotal(IEnumerable<RefundDao> refunds)
     {
         decimal total = 0;
```
Comment (titasstank) [inline Looms.PoS/Looms.PoS.Application/Services/RefundService.cs:10]:
```suggestion
        var total = 0m;
```

**Review by titasstank — CHANGES_REQUESTED** [review]

**Review by titasstank — APPROVED** [review]

**Pull request merged**